core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -62,6 +73,32 @@ oppia.directive('supplementalCard', [
             $scope.helpCardHasContinueButton = false;
           };
 
+          $scope.consumeHint = function() {
+            $scope.helpCardHtml = HintManagerService.getCurrentHint();
+            if (!HintManagerService.areAllHintsExhausted()) {
+              playerTranscriptService.addNewInput(
+                HINT_REQUEST_STRING_I18N_IDS[Math.floor(
+                  Math.random() * HINT_REQUEST_STRING_I18N_IDS.length)], true);
+              $timeout(function () {
+                playerTranscriptService.addNewResponse(
+                  HintManagerService.getCurrentHint());
+                HintManagerService.consumeHint();
```

Comment:
Have consumeHint() return the current hint. Then do:

    var hint = HintManagerService.consumeHint();
    playerTranscriptService.addNewResponse(hint);

Ditto in the tutor card.